modules/core/src/persistence.cpp

```diff
@@ -243,6 +256,13 @@ typedef struct CvFileStorage
     std::deque<char>* outbuf;
 
     base64::Base64Writer * base64_writer;
+    bool is_default_using_base64;
+    base64::fs::State state_of_writing_base64;  /**< used in WriteRawData only */
+
+    bool is_write_struct_delayed;
+    char* delayed_struct_key;
```

Comment:
let's use std::string for these things too.
use std::string::empty() to check whether the strings are empty